backends/platform/sdl/posix/posix.cpp

```diff
@@ -27,6 +27,9 @@
 #define FORBIDDEN_SYMBOL_EXCEPTION_system
 #define FORBIDDEN_SYMBOL_EXCEPTION_random
 #define FORBIDDEN_SYMBOL_EXCEPTION_srandom
+#ifdef __EMSCRIPTEN__
```

Comment:
I might have misunderstood something, but the consensus in #3686 seemed to have been to not add another backend class. So I'm not sure what is preferred.

Reply:
The issue with that PR is that it that it duplicates a lot of stuff from the POSIX backend that looks like it wouldn't be useful on Emscripten (namely the path handling, the Linux AudioCD code and the use of fork/exec for opening the log file). Since having multiple subclasses of `OSystem_SDL` is intended for this exact purpose, I'd recommend one of the following approaches:
- If the POSIX-specific code is useful for Emscripten, then the Emscripten backend can inherit from `OSystem_POSIX` and override specific functions where needed.
- If the POSIX-specific code doesn't work or is overridden by the JavaScript code, then the Emscripten back end can inherit directly from `OSystem_SDL`, with simplified versions of `displayLogFile`, `addSysArchivesToSearchSet`, `getScreenshotsPath`, `getDefaultConfigFileName` and `getDefaultLogFileName` where necessary.

Reply:
Totally agree with @ccawley2011's suggestions

Reply:
Thank you, I'll subclass `OSystem_POSIX` as most POSIX stuff can be reused - Emscripten is generally considered to be POSIX (and says so on [emscripten.org](https://emscripten.org)).

Reply:
I pushed a change implementing a new backend. Let me know what you think.